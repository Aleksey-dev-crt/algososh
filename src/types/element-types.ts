import { ReactElement } from "react";
import { ElementStates } from "./element-states";

export type TSymbol = {
    symbol: string;
    state: ElementStates;
};

export type TElement = {
    element: number;
    state: ElementStates;
};

export type TStackElement = {
    element: string;
    state: ElementStates;
    top: string;
};

export type TQueueElement = {
    element: string;
    state: ElementStates;
    head: string;
    tail: string;
};

export type TListElement = {
    element: string;
    state: ElementStates;
    head: string | ReactElement;
    tail: string | ReactElement;
};