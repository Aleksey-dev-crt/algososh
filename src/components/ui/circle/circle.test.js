import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from "../../../types/element-states";

describe('Circle', () => {
	it('Кружок без буквы рендерится без ошибок', () => {
		const tree = renderer.create(<Circle />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок с буквами рендерится без ошибок', () => {
		const tree = renderer.create(<Circle letter='abcd' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок с head рендерится без ошибок', () => {
		const tree = renderer.create(<Circle head='head' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок с реакт элементом в head рендерится без ошибок', () => {
		const tree = renderer.create(<Circle head={<Circle isSmall={true} letter='head' />} />).toJSON();
		expect(tree).toMatchSnapshot();
	});	
	it('Кружок с tail рендерится без ошибок', () => {
		const tree = renderer.create(<Circle tail='tail' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок с реакт элементом в tail рендерится без ошибок', () => {
		const tree = renderer.create(<Circle tail={<Circle isSmall={true} letter='tail' />} />).toJSON();
		expect(tree).toMatchSnapshot();
	});	
	it('Кружок с index рендерится без ошибок', () => {
		const tree = renderer.create(<Circle index={0} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок с пропом isSmall рендерится без ошибок', () => {
		const tree = renderer.create(<Circle isSmall={true} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок в состоянии default рендерится без ошибок', () => {
		const tree = renderer.create(<Circle state='default' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок в состоянии changing рендерится без ошибок', () => {
		const tree = renderer.create(<Circle state='changing' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Кружок в состоянии modified рендерится без ошибок', () => {
		const tree = renderer.create(<Circle state='modified' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
