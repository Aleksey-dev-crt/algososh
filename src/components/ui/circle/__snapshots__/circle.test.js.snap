// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Кружок без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок в состоянии changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок в состоянии modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle Кружок с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle Кружок с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    >
      abcd
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок с пропом isSmall рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок с реакт элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle"
        >
          head
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Кружок с реакт элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle"
        >
          tail
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
